### PR TITLE
Fix handling of complexity cost for fragments and interfaces

### PR DESCRIPTION
**Pull request by tkuhlman**

### CODE REVIEW

File: cost.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package graphql
 
-import "github.com/GannettDigital/graphql/language/ast"
+import (
+	"github.com/GannettDigital/graphql/language/ast"
```
Comment (gannett-ggreer) [inline cost.go]: Now you're just messing with us. :)

File: cost_test.go [hunk 2]
```diff
@@ -118,121 +133,307 @@ func TestQueryComplexity(t *testing.T) {
...
 	}{
+		// Note a test with an unused fragment isn't needed as that fails and a invalid query
 		{
 			description: "Simple Query",
-			query: `
-				  query Example($size: Int) {
-					a,
-					b
-				  }`,
+			query: `{
```
Comment (gannett-ggreer) [inline cost_test.go:166]: Nit: These example queries have a lot of extra tabs pushing them way to the right.  You must have your tabsize set really small, like at 2.

**Review by gannett-ggreer — APPROVED** [review]

File: cost.go (hunk 1 above)
Comment (tkuhlman) [reply cost.go]: I blame goland

File: cost_test.go (hunk 2 above)
Comment (tkuhlman) [reply cost_test.go:166]: Here is what happened when debugging I didn't want to go past a ton of cases so I would comment these out in a block. That would push it all back and then when I uncommented it wouldn't quite return everything to the right place. I'll fix it, it is really ugly.

**Pull request merged**